src/main/java/com/google/maps/DirectionsApi.java

```diff
@@ -47,9 +49,10 @@ public static DirectionsApiRequest getDirections(GeoApiContext context,
     return newRequest(context).origin(origin).destination(destination);
   }
 
-  static class Response implements ApiResponse<DirectionsRoute[]> {
+  static class Response implements ApiResponse<DirectionsResult> {
```

Comment:
DirectionsApi.getDirections(...) is a nice shorthand that lets a user jump straight to the data, compared with newRequest(..) that let's them tweak more things.

Can you restrict the break to just DirectionsApi.newRequest(..) instead? e.g. by introducing a FullDirectionsApiRequest subclass?

There's likely be some tradeoffs we need to make, so could warrant a quick chat IRL with the team.